WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+					if ( isset( $newlines[ $token[1][ $i ] ] ) ) {
+						$return .= $token[1][ $i ];
+					}
+				}
+			}
+			$token = next( $tokens );
+		}
+		return $return;
+	}
+
+	/**
+	 * Processes the contents of each theme file.
+	 * Obtain sniff helper data
+	 *
+	 * @param string $themedir directory of theme being checked.
+	 * @param array  $themefiles list of files to be checked.
+	 *
+	 * @return false|array
+	 */
+	public function once_through( $themedir, $themefiles = array() ) {
+		// Initiate the Sniff_helper array.
+		global $sniff_helper;
+		$sniff_helper = array(
+			'theme_data' => array(
+				'name'        => '',
+				'uri'         => '',
+				'author'      => '',
+				'author_uri'  => '',
+				'description' => '',
+				'version'     => '',
+				'license'     => '',
+				'license_uri' => '',
+				'tags'        => '',
+				'text_domain' => '',
+			),
+			'theme_supports' => array(
+				'custom-header' => false,
+				'custom-background' => false,
+				'custom-logo' => false,
+				'post-formats' => false,
+				'featured-images' => false,
+				'featured-image-header' => false,
+				'custom-menu' => false,
+			),
+			'comment_reply' => array(
+				'enqueued' => false,
+				'comment_reply_term' => false,
+			),
+			'comments_pagination' => false,
+			'content_width' => false,
+			'add_editor_style' => false,
+			'avatar_check' => false,
+			'custom_menu_support' => false,
+			'post_pagination' => false,
+			'post_format_support' => false,
+			'post_thumbnail_support' => false,
+			'post_tags_support' => false,
+			'title_tag' => array(
+				'theme_support' => false,
+				'wp_title' => false,
+			),
+			'sidebar_support' => array(
+				'register_sidebar_used' => false,
+				'dynamic_sidebar_used' => false,
+				'widgets_init_used' => false,
+			),
+			'basic_function_calls' => array(
+				'wp_footer' => false,
+				'wp_head' => false,
+				'language_attributes' => false,
+				'charset' => false,
+				'automatic_feed_links' => false,
+				'comments_template' => false,
+				'wp_list_comments' => false,
+				'comment_form' => false,
+				'body_class' => false,
+				'wp_link_pages' => false,
+				'post_class' => false,
+			),
+			'doctype' => false,
+			'index_file_used' => false,
+			'style_file_used' => false,
+			'readme_file_used' => false,
+			'screenshot' => array(
+				'found' => false,
+				'less_than_1200_wide' => false,
+				'less_than_900_high' => false,
+				'aspect_ratio_4_by_3' => false,
+				'details_not_found' => false,
+			),
+			'css_required' => array(
+				'sticky' => false,
+				'bypostauthor' => false,
+				'alignleft' => false,
+				'alignright' => false,
+				'aligncenter' => false,
+				'wp-caption' => false,
+				'wp-caption-text' => false,
+				'gallery-caption' => false,
+				'screen-reader-text' => false,
+			),
+			'textdomains' => array(),
+			'themeslug' => '',
+			'files_not_allowed' => array(),
+		);
+		$this->get_minimum_file_check( $themedir );
```

Comment:
The first step around using the global is to pass the variable. So be something like `$sniff_helper = $this->get_minimum_file_check( $sniff_helper, $themedir );`. In `get_minimum_file_check()` you would return `$sniff_helper`.